File: javascript/basic.mjs
/**
 * Web Reader API — Basic Usage (JavaScript)
 * Run: node basic.mjs
 */

const API_KEY = process.env.RAPIDAPI_KEY || "YOUR_RAPIDAPI_KEY";

async function readUrl(url) {
  const params = new URLSearchParams({ url, output: "json" });
  const response = await fetch(
    `https://web-reader-api.p.rapidapi.com/api/read?${params}`,
    {
      headers: {
        "X-RapidAPI-Key": API_KEY,
        "X-RapidAPI-Host": "web-reader-api.p.rapidapi.com",
      },
    }
  );
  return response.json();
}

const data = await readUrl("https://example.com");

console.log(`Title:   ${data.metadata.title}`);
console.log(`Quality: ${data.quality.score}/100 (${data.quality.grade})`);
console.log(`Lang:    ${data.language.language}`);
console.log(`Engine:  ${data.engine}`);
console.log();
console.log("--- Content ---");
console.log(data.content.slice(0, 500));

File: python/basic.py
"""
Web Reader API — Basic Usage
Fetch a URL and get clean Markdown content.
"""

import os
import requests

API_KEY = os.environ.get("RAPIDAPI_KEY", "YOUR_RAPIDAPI_KEY")
BASE_URL = "https://webreader-ai.p.rapidapi.com"
HEADERS = {
    "X-RapidAPI-Key": API_KEY,
    "X-RapidAPI-Host": "webreader-ai.p.rapidapi.com",
}


def read_url(url):
    """Fetch a URL and return structured content."""
    response = requests.get(
        f"{BASE_URL}/api/read",
        headers=HEADERS,
        params={"url": url, "output": "json"},
    )
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    data = read_url("https://example.com")

    print(f"Title:   {data['metadata']['title']}")
    print(f"Quality: {data['quality']['score']}/100 ({data['quality']['grade']})")
    print(f"Lang:    {data['language']['language']}")
    print(f"Engine:  {data['engine']}")
    print()
    print("--- Content ---")
    print(data["content"][:500])
